sidecar/internal/service/reclaimspace.go

```diff
@@ -61,12 +66,29 @@ func (rs *ReclaimSpaceServer) ControllerReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	var deviceSecret map[string]string
+	var volID string
+	var volAttributes map[string]string
+
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	if pv.Spec.CSI != nil {
+		volID = pv.Spec.CSI.VolumeHandle
+		volAttributes = pv.Spec.CSI.VolumeAttributes
+
+		if pv.Spec.CSI.NodeStageSecretRef != nil {
+			// Get the secrets from the k8s cluster
+			deviceSecret, err = kube.GetSecret(ctx, rs.kubeClient, pv.Spec.CSI.NodeStageSecretRef.Name, pv.Spec.CSI.NodeStageSecretRef.Namespace)
```

Comment:
not all drivers will have nodestage secrets. you might need to check if publish secrets exist.